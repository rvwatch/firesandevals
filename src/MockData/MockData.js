export const mockHouse = [
  {
    url: 'https://www.anapioficeandfire.com/api/houses/81',
    name: "House Corbray of Heart's Home",
    region: '',
    coatOfArms:
      'Three black ravens in flight)',
    words: '',
    titles: ["Lord of Heart's Home", 'King of the Fingers (historical)'],
    seats: ["Heart's Home"],
    currentLord: 'https://www.anapioficeandfire.com/api/characters/681',
    heir: 'https://www.anapioficeandfire.com/api/characters/677',
    overlord: 'https://www.anapioficeandfire.com/api/houses/7',
    founded: '',
    founder: 'https://www.anapioficeandfire.com/api/characters/256',
    diedOut: '',
    ancestralWeapons: ['Lady Forlorn'],
    cadetBranches: [],
    swornMembers: [
      'https://www.anapioficeandfire.com/api/characters/255',
      'https://www.anapioficeandfire.com/api/characters/256',
      'https://www.anapioficeandfire.com/api/characters/447',
      'https://www.anapioficeandfire.com/api/characters/654',
      'https://www.anapioficeandfire.com/api/characters/677',
      'https://www.anapioficeandfire.com/api/characters/681',
      'https://www.anapioficeandfire.com/api/characters/871',
      'https://www.anapioficeandfire.com/api/characters/1182'
    ]
  },
  {
    url: 'https://www.anapioficeandfire.com/api/houses/99',
    name: 'House Dayne of Starfall',
    region: 'Dorne',
    coatOfArms:
      'A white sword and falling star crossed on lilac',
    words: '',
    titles: ['Lord of Starfall', 'King of the Torrentine (historical)'],
    seats: ['Starfall'],
    currentLord: 'https://www.anapioficeandfire.com/api/characters/348',
    heir: '',
    overlord: 'https://www.anapioficeandfire.com/api/houses/285',
    founded: '',
    founder: '',
    diedOut: '',
    ancestralWeapons: ['Dawn'],
    cadetBranches: ['https://www.anapioficeandfire.com/api/houses/98'],
    swornMembers: [
      'https://www.anapioficeandfire.com/api/characters/80',
      'https://www.anapioficeandfire.com/api/characters/142',
      'https://www.anapioficeandfire.com/api/characters/151',
      'https://www.anapioficeandfire.com/api/characters/334',
      'https://www.anapioficeandfire.com/api/characters/348',
      'https://www.anapioficeandfire.com/api/characters/1060',
      'https://www.anapioficeandfire.com/api/characters/2121'
    ]
  }
];