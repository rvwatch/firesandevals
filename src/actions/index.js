export const addHouses = houses => ({
  type: 'ADD_HOUSES',
  houses
});


export const addSwornMembers = sworn => ({
  type: 'ADD_SWORN_MEMBERS',
  sworn
});