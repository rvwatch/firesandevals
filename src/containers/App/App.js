import React, { Component } from 'react';
import PropTypes, { shape, func, string } from 'prop-types';
import logo from './logo.svg';
import './App.css';
import { connect } from 'react-redux';
import { addHouses } from '../../actions';
import {fetchData} from '../../ApiCalls/fetchData';
class App extends Component {

  async componentDidMount() {
    // make the initial fetch call here

    console.log('in component did mount');
    const houses = await fetchData();
    console.log(houses)
    this.props.addHouses(houses)
  }

  render() {
    return (
      <div className='App'>
        <div className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <h2>Welcome to Westeros</h2>
          <button onClick={() => {
            this.props.fakeAction();
            alert(this.props.fake);
          }}> FAKE ACTION</button>
        </div>
        <div className='Display-info'>
        
        </div>
      </div>
    );
  }
}

App.propTypes = {
  houses: shape({ houses: string }),
  addHouses: func.isRequired
};

const mapStateToProps = ({ houses }) => ({ houses });
const mapDispatchToProps = dispatch => ({ addHouses:
  (houses) => dispatch(addHouses(houses))
});
export default connect(mapStateToProps, mapDispatchToProps)(App);
