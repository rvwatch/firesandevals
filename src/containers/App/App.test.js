import React from 'react';
import {shallow} from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import * as Actions from '../../actions/';
import * as mock from '../../MockData/MockData.js';

describe('App', () => {

  it('should match the snapshot', () => {
    const addHouses = jest.fn()
    const wrapper = shallow(<App addHouses={addHouses} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should add houses to props', () => {
      const houses = mock.mockHouse;
      const expected = mock.mockHouse;
      const mockState = {houses};
      const mapped = mapStateToProps(mockState);
      expect(mapped.houses).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with correct params for addHouses', () => {
      const mockDispatch = jest.fn();
      const expected = Actions.addHouses(mock.mockHouse);
      const mapped = mapDispatchToProps(mockDispatch);
      mapped.addHouses(mock.mockHouse);
      expect(mockDispatch).toHaveBeenCalledWith(expected)
    }); 
  });
});

