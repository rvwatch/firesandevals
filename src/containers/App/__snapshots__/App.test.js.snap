// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addHouses={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
          />
          <h2>
                    Welcome to Westeros
          </h2>
</div>,
        <div
          className="Display-info"
>
          <div
                    className="wolf"
          />
          <Connect(Component) />
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
/>,
            <h2>
              Welcome to Westeros
</h2>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Westeros",
            },
            "ref": null,
            "rendered": "Welcome to Westeros",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="wolf"
/>,
            <Connect(Component) />,
          ],
          "className": "Display-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "wolf",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="App-header"
>
            <img
                        alt="logo"
                        className="App-logo"
                        src="logo.svg"
            />
            <h2>
                        Welcome to Westeros
            </h2>
</div>,
          <div
            className="Display-info"
>
            <div
                        className="wolf"
            />
            <Connect(Component) />
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
/>,
              <h2>
                Welcome to Westeros
</h2>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to Westeros",
              },
              "ref": null,
              "rendered": "Welcome to Westeros",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="wolf"
/>,
              <Connect(Component) />,
            ],
            "className": "Display-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "wolf",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
